river/command/input.zig

```diff
@@ -0,0 +1,108 @@
+// This file is part of river, a dynamic tiling wayland compositor.
+//
+// Copyright 2021 The River Developers
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+const std = @import("std");
+const mem = std.mem;
+
+const server = &@import("../main.zig").server;
+const util = @import("../util.zig");
+
+const Error = @import("../command.zig").Error;
+const Seat = @import("../Seat.zig");
+const InputConfig = @import("../InputConfig.zig");
+const InputManager = @import("../InputManager.zig");
+
+pub fn input(
+    allocator: *mem.Allocator,
+    seat: *Seat,
+    args: []const []const u8,
+    out: *?[]const u8,
+) Error!void {
+    if (args.len < 4) return Error.NotEnoughArguments;
+    if (args.len > 4) return Error.TooManyArguments;
+
+    var new_input_config_node: ?*std.SinglyLinkedList(InputConfig).Node = null;
+    var input_config: ?*InputConfig = null;
+
+    // Try to find an existing InputConfig with matching identifier.
+    var conf_it = server.input_manager.input_configs.first;
+    while (conf_it) |conf_node| : (conf_it = conf_node.next) {
+        if (mem.eql(u8, conf_node.data.identifier, args[1])) {
+            input_config = &conf_node.data;
+            break;
+        }
+    }
+
+    // If no matching InputConfig has been found, create one.
+    if (input_config == null) {
```

Comment:
Instead of making `input_config` optional and checking if it is null here, it would be more idiomatic to use the else clause of the while loop to handle the case in which it does not exist:
```
const input_config = while (...) {
} else blk: {
     // create input config
     break :blk input_config;
};
```
with this approach the `.?` on `input_config` throughout the rest of this function can be avoided.